HARK/distributions/base.py

```diff
@@ -3,22 +3,39 @@
 import numpy as np
 from numpy import random
 
+MAX_INT32 = 2**31 - 1
+
+
+def random_seed():
+    """
+    Generate a random seed for use in random number generation. This random seed
+    is derived from the system clock and other variables, and is therefore
+    different every time the code is run.
+    For discussion on random number generation and random seeds, see
+    https://docs.scipy.org/doc/scipy/tutorial/stats.html#random-number-generation
+    Parameters
+    ----------
+    None
+    Returns
+    -------
+    seed : int
+        Random seed.
+    """
+    return random.SeedSequence().entropy
+
 
 class Distribution:
     """
     Base class for all probability distributions
     with seed and random number generator.
 
-    For discussion on random number generation and random seeds, see
-    https://docs.scipy.org/doc/scipy/tutorial/stats.html#random-number-generation
-
     Parameters
     ----------
     seed : Optional[int]
         Seed for random number generator.
     """
 
-    def __init__(self, seed: Optional[int] = 0) -> None:
+    def __init__(self, seed: Optional[int] = None) -> None:
```

Comment:
Breaking change: default seed changed from `0` to `None`. Code relying on the previous default will no longer be reproducible.